# SC-shapes-the-maturation-of-cortical-morphology
## Code:
1.	**Spin test**: The spin test was conducted using an open Matlab code package (https://github.com/frantisekvasa/rotate_parcellation) (1, 2).
2.	**Rewired test**: Surrogate networks that preserve the nodal degree and approximate edge length distribution of the empirical network were generated using a Matlab code (https://www.brainnetworkslab.com/coderesources) (3). 
3.	**Group SC generation**: Group-level connectome backbone was generated by using a consensus approach that preserves the connection length distributions of individual networks (https://www.brainnetworkslab.com/coderesources) (4).
4.	**Mixed linear model**: mixed_model_group_diff.m
5.	**White matter network constraints on the spatial maturation of cortical thickness**: Liang_corrNeighborCT.m
Liang_corrNeighborCT_regdist.m
6.	**Random walk model**: Liang_ramdom_walk.m
7.	**SVR model**: The CT maturation degree was predicted with an SVR model using an open Matlab code package (https://github.com/ZaixuCui/Pattern_Regression_Clean/tree/master/SVR) (5).
8.	**Computing transcription level differences for 4 neurodevelopmental gene sets (6) between dominant and non-dominant regions**: BrainSpan_dominant.m
9.	**Preprocessing of AHBA gene expression data**: The regional microarray expression data were preprocessed using a recommended pipeline with the abagen toolbox (https://github.com/rmarkello/abagen) (7, 8).
AHBA_gene_expression.py
10.	**Gene Ontology Enrichment Analysis**: Gene Ontology enrichment analysis was conducted using the ToppGene Suite (https://toppgene.cchmc.org/) (9).

## Data：
1.	Multiscale Desikan-Kiliany parcellation files (10) were downloaded using netneurotools toolbox (https://github.com/netneurolab/netneurotools).
2.	Maturation extents and significances of cortical thickness at each parcellation resolution: 
TVector.txt, PVector.txt
3.	Group-level connectome backbone at each parcellation resolution:
Group_sc.mat, Group_sc_nonei.mat (For each node, we excluded all its spatially adjoining neighbors.)
4.	Matrix of the Euclidean distance between every two nodes:
dist.mat
5.	The tissue samples in Brainspan dataset belong to dominant regions:
dominant_regions.mat (‘0’ and ‘1’ denote non-dominant and dominant regions.)
6.	Neurodevelopmental processes related genes (6): neurodev_process.mat

## References
1.	F. Vasa et al., Adolescent Tuning of Association Cortex in Human Structural Brain Networks. Cereb Cortex 28, 281-294 (2018).
2.	A. F. Alexander-Bloch et al., On testing for spatial correspondence between maps of human brain structure and function. Neuroimage 178, 540-551 (2018).
3.	R. F. Betzel, D. S. Bassett, Specificity and robustness of long-distance connections in weighted, interareal connectomes. Proceedings of the National Academy of Sciences 115, E4880-E4889 (2018).
4.	R. F. Betzel, A. Griffa, P. Hagmann, B. Mišić, Distance-dependent consensus thresholds for generating group-representative structural brain networks. Network neuroscience 3, 475-496 (2019).
5.	Z. Cui, G. Gong, The effect of machine learning regression algorithms and sample size on individualized behavioral prediction with functional connectivity features. Neuroimage 178, 622-637 (2018).
6.	H. J. Kang et al., Spatio-temporal transcriptome of the human brain. Nature 478, 483-489 (2011).
7.	R. D. Markello et al., Standardizing workflows in imaging transcriptomics with the abagen toolbox. Elife 10, e72129 (2021).
8.	A. Arnatkevic̆iūtė, B. D. Fulcher, A. Fornito, A practical guide to linking brain-wide gene expression and neuroimaging data. Neuroimage 189, 353-367 (2019).
9.	J. Chen, E. E. Bardes, B. J. Aronow, A. G. Jegga, ToppGene Suite for gene list enrichment analysis and candidate gene prioritization. Nucleic acids research 37, W305-W311 (2009).
10.	L. Cammoun et al., Mapping the human connectome at multiple scales with diffusion spectrum MRI. Journal of neuroscience methods 203, 386-397 (2012).
11.	J. A. Miller et al., Transcriptional landscape of the prenatal human brain. Nature 508, 199-206 (2014).

